src/common/middleware/call-api.js

```diff
@@ -0,0 +1,101 @@
+import { checkStatus, getError } from '../helpers/api';
+
+// Action key that carries API call info interpreted by this Redux middleware.
+export const CALL_API = 'CALL_API';
+
+// A Redux middleware that interprets actions with CALL_API info specified.
+// Performs the call and promises when such actions are dispatched.
+export default defaults => () => next => async action => {
```

Comment:
A small variation on the Redux real-world example is an extra function around the middleware to memoise default settings, such as the API basepath (`defaults.endpoint`).